pinecone/client_test.go

```diff
@@ -313,6 +313,29 @@ func (ts *IntegrationTests) TestGenerateEmbeddingsInvalidInputs() {
 	require.Contains(ts.T(), err.Error(), "TextInputs must contain at least one value")
 }
 
+func (ts *IntegrationTests) TestRerankDocument() {
+	ctx := context.Background()
+	rerankModel := "bge-reranker-v2-m3"
+	topN := 4
+	retunDocuments := true
+	ranking, err := ts.client.Inference.Rerank(ctx, &RerankRequest{
+		Model:           rerankModel,
+		Query:           "i love cats",
+		ReturnDocuments: &retunDocuments,
+		TopN:            &topN,
+		RankFields:      &[]string{"text"},
+		Documents: []Document{
```

Comment:
Note: this is the situation in which the doc will only be ranked on `text`, in case the user isn't clear on that re: my comment [here](https://github.com/pinecone-io/go-pinecone/pull/80/files#r1805126027)